src/androidTest/java/com/couchbase/lite/DatabaseEncryptionTest.java

```diff
@@ -230,4 +234,67 @@ public boolean update(UnsavedRevision rev) {
         Assert.assertNotNull(seekrit);
         Assert.assertEquals(1, seekrit.getDocumentCount());
     }
+
+    public void testEncryptedAttachments() throws Exception {
+        if (!isSQLiteDB())
+            return;
+
+        Manager cryptoManager = getEncryptionTestManager();
+        cryptoManager.registerEncryptionKey("letmein", "seekrit");
+        Database seekrit = cryptoManager.getDatabase("seekrit");
+        Assert.assertNotNull(seekrit);
+
+        // Save a doc with an attachment:
+        Document doc = seekrit.getDocument("att");
+        byte[] body = "This is a test attachment!".getBytes();
+        ByteArrayInputStream is = new ByteArrayInputStream(body);
+        UnsavedRevision rev = doc.createRevision();
+        rev.setAttachment("att.txt", "text/plain; charset=utf-8", is);
+        SavedRevision savedRev = rev.save();
+        Assert.assertNotNull(savedRev);
+
+        // Read the raw attachment file and make sure it's not clear text:
+        Map<String, Object> atts = (Map<String, Object>)savedRev.getProperties().get("_attachments");
+        Map<String, Object> att = (Map<String, Object>)atts.get("att.txt");
+        String digest = (String) att.get("digest");
+        Assert.assertNotNull(digest);
+
+        BlobKey key = new BlobKey(digest);
+        String path = seekrit.getAttachmentStore().getRawPathForKey(key);
+        byte[] raw = TextUtils.read(new FileInputStream(path));
+        Assert.assertNotNull(raw);
+        Assert.assertTrue(!Arrays.equals(raw, body));
+    }
+
+    public void testRekey() throws Exception {
+        if (!isSQLiteDB())
+            return;
+
+        // First run the encrypted-attachments test to populate the db:
+        testEncrypedAttachments();
+
+        Manager cryptoManager = getEncryptionTestManager();
+        Database seekrit = cryptoManager.getDatabase("seekrit");
+        seekrit.changeEncryptionKey("letmeout");
+
+        // Close & reopen seekrit:
+        String dbName = seekrit.getName();
+        Assert.assertTrue(seekrit.close());
+        seekrit = null;
+
+        cryptoManager.registerEncryptionKey("letmeout", "seekrit");
+        Database seekrit2 = cryptoManager.getDatabase("seekrit");
+        Assert.assertNotNull(seekrit2);
+        seekrit = seekrit2;
+
+        // Check the document and its attachment:
+        SavedRevision savedRev = seekrit.getDocument("att").getCurrentRevision();
+        Assert.assertNotNull(savedRev);
+
+        Attachment att = savedRev.getAttachment("att.txt");
+        Assert.assertNotNull(att);
+        byte[] body = "This is a test attachment!".getBytes();
+        byte[] rawAtt = TextUtils.read(att.getContent());
+        Assert.assertTrue(Arrays.equals(rawAtt, body));
```

Comment:
Let me clarify more with the code:

// Here is the text content of the attachment converted to raw bytes:
byte[] body = "This is a test attachment!".getBytes();

// Here is the decrypted raw bytes read from the attachment.
byte[] rawAtt = TextUtils.read(att.getContent());

// There is an assertion to check if it's the same raw bytes
Assert.assertTrue(Arrays.equals(rawAtt, body));

// Here is your recommendation which does the same thing as above because they all are decrypted raw bytes:
Assert.assertTrue(Arrays.equals(rawAtt, prvRawAtt));